### ISSUE TITLE
Inconsistent behaviour when escaping wildcards with backslash

### ISSUE THREAD

**Issue by sophiakoulen** [sophiakoulen/minishell#126]
`echo \*` prints `\*`, which should be just `*`.
I know we do not have to interpret backslashes, but in our implementation backslashes can escape $, quotes and whitespaces already.
If we choose to add the backslash feature, it is important that we do it consistently, else our "feature" (which is not asked for) could be seen as a bug.

**Comment by sophiakoulen** [comment]
Similarly, if we handle backslash escaping, we should be able to escape characters like `|` , `<`, `&` during tokenization. This shouldn't be hard to do if we reuse the update_state function.

**Comment by znichola** [comment]
yes, this follows, and makes sense. But I will take the library of also adding the `side-quest` label.
